src/NJsonSchema/Infrastructure/XmlObjectExtension.cs

```diff
@@ -29,14 +29,35 @@ public static void GenerateXmlObjectForType(this JsonSchema4 schema, Type type)
                     GenerateXmlObject(xmlTypeAttribute.TypeName, xmlTypeAttribute.Namespace, false, false, schema);
             }
         }
+
+        /// <summary>
+        /// Generate XML object for a JSON Schema definition
+        /// </summary>
+        /// <param name="schema">The JSON Schema</param>
+        /// <param name="type">The array type</param>
+        public static void GenerateXmlObjectForArrayType(this JsonSchema4 schema, Type type)
+        {
+            if (schema.IsArray && schema.ParentSchema == null)
```

Comment:
Why `schema.ParentSchema == null`? I think it is not completely safe to rely on ParentSchema...

Reply:
What i am after mainly is to detect if it is the first parameter to the controller or not, e.g. not a DTO property of type Array. I it is, the ArrayOf[Name] should be applied. Do you see any other way to detect that?

Reply:
So it's to detect if it's a direct usage in a parameter vs. part of a schema (e.g. property)? I think for this case, checking the ParentSchema is fine... let me know if you see problems in your app with this approach (I personally don't use xml schemas in any project).

Reply:
Correct, thats the difference. Earlier I assumed that all arrays were properties on a DTO/Schema which is not correct....